alibi_detect/base.py

```diff
@@ -229,6 +225,17 @@ def from_config(self): ...
     def _set_config(self): ...
 
 
+@runtime_checkable
+class StatefulDetectorOnline(ConfigurableDetector, Protocol):
+    """Type Protocol for detectors that have support for save/loading of online state.
+
+    Used for typing save and load functionality in `alibi_detect.saving.saving`.
+    """
+    def save_state(self, filepath): ...
+
+    def load_state(self, filepath): ...
```

Comment:
Should parameters have type hints?

Reply:
Added in 15d1dfaf49ab2a34eb82d78b39449992ee210d14. Note: I also updated the pre-existing `get_config` and `set_config` methods here.